email/mock.go

```diff
@@ -38,12 +43,26 @@ func (m *MockSender) GetNumSentEmails() int {
 	return len(m.Emails)
 }
 
-// GetEmailOnIndex will return nth sent email, n is sent by parameter
-// it may return error if index is out of range
-func (m *MockSender) GetEmailOnIndex(n int) (Record, error) {
-	if n >= m.GetNumSentEmails() {
-		return Record{}, errors.New("Index out of range")
-	}
+// GetEmail will return email on top of the recording channel
+// this will block the program if no email is present
+func (m *MockSender) GetEmail() Record {
+	return <-m.Emails
+}
 
-	return m.Emails[n], nil
+// SetMaxEmailsAndReset will set a maximum of stored emails
+// to value given by parameter and reset stored emails
+func (m *MockSender) SetMaxEmailsAndReset(MaxEmails int) {
+	close(m.Emails)
+	m.maxRecordedEmails = MaxEmails
+	m.Emails = make(chan Record, MaxEmails)
+}
+
+// Reset will delete all emails present in MockSender
+func (m *MockSender) Reset() {
+	close(m.Emails)
+	if m.maxRecordedEmails != 0 {
+		m.Emails = make(chan Record)
```

Comment:
`make(chan Record, 0)` and `make(chan Record)` should be the same. Can't see why we need if here